File: Leetcode/299 - Bulls and Cows/solution.js
var getHint = function (secret, guess) {
    let bulls = 0;
    const secretMap = {}, guessMap = {};

    for (let i = 0; i < secret.length; i++) {
        if (secret[i] === guess[i]) {
            bulls++;
        } else {
            secretMap[secret[i]] = (secretMap[secret[i]] ?? 0) + 1;
            guessMap[guess[i]] = (guessMap[guess[i]] ?? 0) + 1;
        }
    }

    let cows = 0;
    for (const ch in secretMap) {
        if (guessMap[ch]) {
            cows += Math.min(secretMap[ch], guessMap[ch]);
        }
    }

    return `${bulls}A${cows}B`;
};

File: Leetcode/299 - Bulls and Cows/solution.cpp
string getHint(string secret, string guess)
{
    int bulls = 0;
    unordered_map<char, int> secretMap, guessMap;
    for (int i = 0; i < secret.size(); i++)
    {
        if (secret[i] == guess[i])
        {
            bulls++;
        }
        else
        {
            secretMap[secret[i]]++;
            guessMap[guess[i]]++;
        }
    }

    int cows = 0;
    for (auto [key, val] : secretMap)
    {
        cows += min(val, guessMap[key]);
    }
    return to_string(bulls) + "A" + to_string(cows) + "B";
}